cmd/api/main.go

```diff
@@ -40,9 +131,42 @@ func main() {
 		log.Fatal("Failed to migrate database:", err)
 	}
 
+	// Initialize auth
+	if err := auth.InitAuth(); err != nil {
+		log.Printf("Warning: OAuth initialization failed: %v", err)
+	}
+
 	// Initialize router
 	r := gin.Default()
 
+	// Add CORS middleware with more permissive settings for development
+	corsConfig := cors.Config{
+		AllowOrigins:     []string{cfg.FrontendURL},
+		AllowMethods:     []string{"GET", "POST", "PUT", "DELETE", "OPTIONS"},
+		AllowHeaders:     []string{"Origin", "Content-Type", "Accept", "Authorization", "Cookie"},
+		ExposeHeaders:    []string{"Set-Cookie"},
+		AllowCredentials: true,
+		MaxAge:           12 * time.Hour,
+	}
+	r.Use(cors.New(corsConfig))
+
+	// Setup session store with development-friendly settings
+	store, err := setupStore()
+	if err != nil {
+		log.Fatal("Failed to setup session store:", err)
+	}
+
+	// Modify cookie settings for development
+	store.Options(sessions.Options{
+		Path:     "/",
+		MaxAge:   86400 * 7,
+		HttpOnly: true,
+		Secure:   false,                // Set to false for development
+		SameSite: http.SameSiteLaxMode, // Use Lax for development
+	})
```

Comment:
Maybe create env variable config.DevelopmentMode and have env variable `DEVELOPMENT=True` in order for these to be set?